src/ui/map.js

```diff
@@ -1412,7 +1412,6 @@ class Map extends Camera {
         try {
             if (this.style.setState(style)) {
                 this._update(true);
-                this._updateTerrain();
```

Comment:
Removing this is fine because `_update` calls `setTerrain` since it performs a style diff internally.